psi4/driver/p4util/solvers.py

```diff
@@ -304,3 +308,742 @@ def extrapolate(self, out=None):
             out.axpy(c, self.state[num])
 
         return out
+
+
+def _diag_print_heading(title_lines, solver_name, max_ss_size, nroot, e_tol, r_tol, maxiter, verbose=1):
+    """Print a message to the output file when the solver has processed all options and is ready to begin"""
+    if verbose < 1:
+        # no printing
+        return
+    # show title if not silent
+    core.print_out("\n\n")
+    core.print_out("\n".join([x.center(77) for x in title_lines]))
+    core.print_out("\n")
+    if verbose > 1:
+        # summarize options for verbose
+        core.print_out("   " + "{} options".format(solver_name) + "\n")
+        core.print_out("\n  -----------------------------------------------------\n")
+        core.print_out("    Maxiter                         = {:<5d}\n".format(maxiter))
+        core.print_out("    Eigenvalue tolerance            = {:11.5e}\n".format(e_tol))
+        core.print_out("    Eigenvector tolerance           = {:11.5e}\n".format(r_tol))
+        core.print_out("    Max number of expansion vectors = {:<5d}\n".format(max_ss_size))
+        core.print_out("\n")
+    # show iteration info headings if not silent
+    core.print_out("  => Iterations <=\n")
+    if verbose == 1:
+        # default printing one line per iter max delta value and max residual norm
+        core.print_out("  {}           {}      {}\n".format(" " * len(solver_name), "Max[D[value]]", "Max[|R|]"))
+    else:
+        # verbose printing, value, delta, and |R| for each root
+        core.print_out("    {}       {}      {}      {}\n".format(" " * len(solver_name), "value", "D[value]", "|R|"))
```

Comment:
In future probably want to print to a string (or a list that can be `'\n'.join(textlist)` at the end) so that results can go to screen, outputfile, or logging easily.